src/service/async_service_impl.h

```diff
@@ -263,7 +228,6 @@ class AsyncAuthServiceImpl {
 
   envoy::service::auth::v2::Authorization::AsyncService service_v2_;
   envoy::service::auth::v3::Authorization::AsyncService service_;
-  grpc::health::v1::Health::AsyncService health_service_;
```

Comment:
Instead of  calling `new HealthCheckServer` in the `Run` directly, can we keep making the server as a field owned by the AsyncAuthserviceImpl? This way it's clearer the class has the health check functionality and object ownership (as opposed to new).